tutorial/LFRic/background/LFRic_PSy.md

```diff
@@ -0,0 +1,206 @@
+      nlayers = field_w0_proxy%vspace%get_nlayers()
+      !
+      ! Look-up dofmaps for each function space
+      !
+      map_w0 => field_w0_proxy%vspace%get_whole_dofmap()
+      !
+      ! Initialise number of DoFs for w0
+      !
+      ndf_w0 = field_w0_proxy%vspace%get_ndf()
+      undf_w0 = field_w0_proxy%vspace%get_undf()
+```
+
+Finally, there is a generated call to `setval_field_w0_code` in the
+`setval_field_w0_kernel_mod` with the complete argument list
+```fortran
+      !
+      ! Call our kernels
+      !
+      DO cell=1,field_w0_proxy%vspace%get_ncell()
+        !
+        CALL setval_field_w0_code(nlayers, field_w0_proxy%data, &
+             scalar_w0, ndf_w0, undf_w0, map_w0(:,cell))
+      END DO
+```
+
+with the same argument order as in the [completed kernel example](
+../example1/solutions/part1/setval_field_w0_kernel_mod.f90)
+provided in the [Example 1, solution of Part 1](
+../example1/solutions/part1) (generated by running `genkernelstub`).
+
+The above kernel call code call illustrates how `iterates_over = CELLS`
+metadata in the [`setval_field_w0_kernel_mod.f90`](
+../example1/solutions/part1/setval_field_w0_kernel_mod.f90)
+translates to a **loop over cells** (`cell` loop counter) in the PSy layer.
+
+## Generated code to support built-in calls
+
+As illustrated in the [*`invoke` calls* section](
+LFRic_algorithm.md#invoke-calls) of algorithm introduction,
+the `invoke` call from the completed algorithm in
+[Example 2 solution](../example2/solutions/example2_alg_mod.x90)
+
+```fortran
+    call invoke( name = "Builtins on W0 and W3 fields",              &
+                 ...
+                 X_plus_Y(field_out_w0, field1_in_w0, field2_in_w0), &
+                 ... )
+```
+
+translates to this PSy-layer subroutine call
+
+```fortran
+    CALL invoke_builtins_on_w0_and_w3_fields(field_out_w0, field_out_w3, &
+         field1_in_w0, field2_in_w0, field1_in_w3, field2_in_w3)
+```
+
+in the generated `example2_alg_mod.f90` file (not kept in the repository).
+
+The generated `invoke_builtins_on_w0_and_w3_fields` subroutine code
+is located in the `example2_alg_mod_psy.f90` file (also not kept in the
+repository).
+
+As for a kernel call there are, amongst others, generated declarations and
+calls to the `field_out_w0`, `field1_in_w0` and `field2_in_w0` proxies
+
+```fortran
+      TYPE(field_proxy_type) field_out_w0_proxy, field_out_w3_proxy, &
+                             field1_in_w0_proxy, field2_in_w0_proxy, &
+                             field1_in_w3_proxy, field2_in_w3_proxy
+      ...
+      !
+      ! Initialise field and/or operator proxies
+      !
+      field_out_w0_proxy = field_out_w0%get_proxy()
+      ...
+      field1_in_w0_proxy = field1_in_w0%get_proxy()
+      field2_in_w0_proxy = field2_in_w0%get_proxy()
+```
+
+and generated declarations and dereferencing assignments to access
+the proxy data
+
+```fortran
+      INTEGER(KIND=i_def) undf_aspc1_field_out_w0, undf_aspc1_field_out_w3, &
```

Comment:
I just noticed you're using the common practice of capitalising Fortran keywords. This potentially makes the document more friendly to old-school Fortran programmers but is contrary to actual LFRic policy. Something to consider.

Reply:
This is something that comes from PSyclone so I quoted it as it is. I added a note that according to LFRic coding standards we don't capitalise Fortran keyword in the LFRic code.